### PR TITLE
[Tab Strip]: Fix divider colors

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/40425

| | Before | After |
|--------|--------|--------|
| Light | image | image |
| Dark | image | image |
| Private | image | image |
| Tor | image | image | 

Seems like this only really affected Light mode. Looks like Chromium introduced some new ColorIds for these, and as a result, started ignoring the `GetTabSeparatorColor` in some cases. I completely removed the function, as vertical tabs isn't affected anyway.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See issue

### CODE REVIEW

File: browser/ui/views/tabs/brave_tab_strip.cc [hunk 1]
```diff
@@ -197,33 +197,6 @@ void BraveTabStrip::AddedToWidget() {
   }
 }
 
-SkColor BraveTabStrip::GetTabSeparatorColor() const {
-  if (ShouldShowVerticalTabs()) {
-    return SK_ColorTRANSPARENT;
-  }
```
Comment (sangwoo108) [inline browser/ui/views/tabs/brave_tab_strip.cc:203]: Nice catch, as we sets 0 opacity for vertical tabs, so we don't need this.

File: browser/ui/views/tabs/brave_tab_strip.cc [hunk 2]
```diff
@@ -197,33 +197,6 @@ void BraveTabStrip::AddedToWidget() {
...
-
-    // For incognito/guest window.
-    return SkColorSetRGB(0x3F, 0x32, 0x56);
-  }
-
-  // If custom theme is used, follow upstream separator color.
-  auto* theme_service = ThemeServiceFactory::GetForProfile(profile);
-  if (theme_service->GetThemeSupplier()) {
-    return TabStrip::GetTabSeparatorColor();
-  }
```
Comment (sangwoo108) [inline browser/ui/views/tabs/brave_tab_strip.cc:219]: I guess custom theme should be fine, as the `TabStrip::GetTabSeparatorColor()` picks blended color considering min contrast between the background.

https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/views/tabs/tab_strip.cc;l=2188;drc=b996220ae5827a19279d135b68d105118b466166

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

**Comment by GeetaSarvadnya** [comment]
Verification INPROGRESS on

```
Brave | 1.71.1 Chromium: 128.0.6613.27 (Official Build) nightly (64-bit)
-- | --
Revision | 904084cc7cd349200a575d45212c794fe5d69dfe
OS | Windows 10 Version 22H2 (Build 19045.4780)
```

**Regular window**

Light theme | Dark theme
--------------|----------

**Private window**

Light theme | Dark theme
--------------|----------

**Tor window**

Light theme | Dark theme
--------------|----------

**Pull request merged**